### PR TITLE
PROD-865#2064 - Fix Forum List Widgets Disregarding (Parent Forum ID) on the widget query

### LINKED ISSUE #2064: (BB) Forum List Widgets Disregarding (Parent Forum ID) on the widget query
**Describe the bug**
Any of this option ( "0" to show only root - "any" to show all ) is disregarded by the query. The client is confused by these options. No Difference in the results whether "0" or "any" arguments. i found out that the function widget is disregarding this option.

**To Reproduce**
---

**Expected behavior**
If we continue with this option, we should improve the query according to this option( "0" to show only root - "any" to show all ) or totally remove this option on the forum list widget

**Screenshots**
[https://prnt.sc/xdktie](https://prnt.sc/xdktie)
[https://prnt.sc/xfceh0](https://prnt.sc/xfceh0)
[https://prnt.sc/xfe3p0](https://prnt.sc/xfe3p0)

**Support ticket links**
[https://secure.helpscout.net/conversation/1396840734/119948?folderId=4222127](https://secure.helpscout.net/conversation/1396840734/119948?folderId=4222127)

**Jira issue** : [PROD-865]

[PROD-865]: https://buddyboss.atlassian.net/browse/PROD-865?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### PR DESCRIPTION
**Pull request by kent-carlo**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #2064.

### How to test the changes in this Pull Request:

https://www.loom.com/share/fb483f0633b4409fa618eafce47c9b36

### Proof Screenshots or Video

https://www.loom.com/share/fb483f0633b4409fa618eafce47c9b36

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Fixed forum parent ID so can display all forums using 'any' and display only parent forums using '0' option.

### CODE REVIEW

File: src/bp-forums/common/widgets.php [hunk 1]
```diff
@@ -591,7 +591,7 @@ public function widget( $args, $instance ) {
 			array(
 				'post_type'           => bbp_get_forum_post_type(),
 				// 'post_parent'         => $settings['parent_forum'],
```
Comment (dipskakadiya) [inline src/bp-forums/common/widgets.php:593]: @kent-carlo This is the same line that you added so can you check history why it's removed.

File: src/bp-forums/common/widgets.php [hunk 2]
```diff
@@ -591,7 +591,7 @@ public function widget( $args, $instance ) {
...
-				'post_parent'         => 0,
+				'post_parent'         => intval( $settings['parent_forum'] ),
```
Comment (dipskakadiya) [inline src/bp-forums/common/widgets.php:594]: @kent-carlo I think you need to handle case for `$settings['parent_forum'] = 'any'` here.

**Review by dipskakadiya — CHANGES_REQUESTED** [review]

File: src/bp-forums/common/widgets.php (hunk 1 above)
Comment (kent-carlo) [reply src/bp-forums/common/widgets.php:593]: @dipskakadiya It only says "forums list widgets updates", should I remove it? This is the commit https://github.com/buddyboss/buddyboss-platform/commit/8ac8a622044ad82a20f95293935d9fe194df90bd

**Review by dipskakadiya — APPROVED** [review]

File: src/bp-forums/common/widgets.php [hunk 3]
```diff
@@ -590,8 +590,7 @@ public function widget( $args, $instance ) {
 		$widget_query = new WP_Query(
 			array(
 				'post_type'           => bbp_get_forum_post_type(),
-				// 'post_parent'         => $settings['parent_forum'],
-				'post_parent'         => 0,
+				'post_parent'         => $settings['parent_forum'],
```
Comment (jitendrabanjara1991) [inline src/bp-forums/common/widgets.php:593]: ! empty( $settings['parent_forum'] ) ? $settings['parent_forum'] : 0

**Review by jitendrabanjara1991 — CHANGES_REQUESTED** [review]

**Review by KartikSuthar — APPROVED** [review]

**Pull request merged**